packages/www/pages/dashboard/streams/[id]/index.tsx

```diff
@@ -38,39 +49,131 @@ const Overview = () => {
+  }, [doGetInfo]);
+
+  useEffect(() => {
+    const intervalId = window.setInterval(() => {
+      if (info) {
+        getIngestRate(typeof id === "string" ? id : null);
+      } else return null;
+    }, ingestInterval);
+
+    return () => {
+      window.clearInterval(intervalId);
+    };
+  }, [getIngestRate, info]);
+
   return (
     <StreamDetail
-      activeTab="Overview"
+      activeTab={switchTab}
       stream={stream}
       streamHealth={streamHealth}
       invalidateStream={invalidateStream}
+      setSwitchTab={setSwitchTab}
       breadcrumbs={[
         { title: "Streams", href: "/dashboard/streams" },
         { title: stream?.name },
       ]}>
-      <MultistreamTargetsTable
-        stream={stream}
-        streamHealth={streamHealth}
-        invalidateStream={invalidateStream}
-        css={{ mb: "$7" }}
-        emptyState={
-          <Text variant="gray" size="2">
-            No targets
-          </Text>
-        }
-        tableLayout="auto"
-        border
-      />
-      <StreamSessionsTable
-        streamId={id}
-        emptyState={
-          <Text variant="gray" size="2">
-            No sessions
-          </Text>
-        }
-        tableLayout="auto"
-        border
-      />
+      {switchTab === "Overview" ? (
+        <>
+          <MultistreamTargetsTable
+            stream={stream}
+            streamHealth={streamHealth}
+            invalidateStream={invalidateStream}
+            css={{ mb: "$7" }}
+            emptyState={
+              <Text variant="gray" size="2">
+                No targets
+              </Text>
+            }
+            tableLayout="auto"
+            border
+          />
+          <StreamSessionsTable
+            streamId={id}
+            emptyState={
+              <Text variant="gray" size="2">
+                No sessions
+              </Text>
+            }
+            tableLayout="auto"
+            border
+          />
+        </>
+      ) : (
+        <>
+          <HealthChecksTable
+            stream={stream}
+            streamHealth={streamHealth}
+            invalidateStream={invalidateStream}
+            css={{ mb: "$7" }}
+            emptyState={
+              <Text variant="gray" size="2">
+                No targets
+              </Text>
+            }
+            tableLayout="auto"
+            border
+          />
+          <Logger stream={stream} css={{ mb: "$7" }} />
+          <Box
+            css={{
+              borderBottom: "1px solid",
+              borderColor: "$mauve6",
+              pb: "$2",
+              mb: "$7",
+              width: "100%",
+            }}>
+            <Heading size="1" css={{ fontWeight: 500, mb: "$1" }}>
+              Session ingest rate
+            </Heading>
+            <Text variant="gray" size="3">
+              After the stream loads, ingest rate updates every 10 seconds.
+            </Text>
+          </Box>
+          <Chart data={dataChart} />
+        </>
```

Comment:
Here, instead of copying the implementation of the stream health tab, we could just insert a `StreamHealthTab` component or something like that! WDYT?

Ideally we would also have a separate component for the `"Overview"` tab, and this whole page here would only need to worry about the tab-switching logic. That would be awesome, but don't feel obliged to do that refactor. Just keeping the stream health tab separate, as it was before, already LGTM.